src/Utils.cc

```diff
@@ -133,3 +133,19 @@ utils::firstChar(const QString &input)
 
         return QString::fromUcs4(&input.toUcs4().at(0), 1).toUpper();
 }
+
+QString
+utils::humanReadableFileSize(const qlonglong bytes)
+{
+        constexpr static const char *units[] = {"B", "KB", "MB", "GB", "TB"};
+        constexpr static const int length    = sizeof(units) / sizeof(units[0]);
+
+        int u       = 0;
+        double size = static_cast<double>(bytes);
+        while (size >= 1024.0 && u < length) {
+                ++u;
+                size /= 1024.0;
+        }
+
+        return QString::number(size, 'f', 2) + ' ' + units[u];
```

Comment:
Use `'g'` formatting and 4 significant digits. This provides a consistent level of precision and reduces variation in string length, rather than just unconditionally slapping on two decimal places no matter the order of magnitude.

Reply:
Noted and corrected. Thanks!